### DISCUSSION TITLE
Mushr in Gazebo

### DISCUSSION THREAD

**Discussion by ArguiIm** [prl-mushr/mushr#73] [General]
Hello, 
I'm new to ROS and Gazebo. 
After following the tutorials, I needed to use mushr robots in Gazebo. Until now, I have the robot in Gazebo, but the simulator can't detect the front wheels, and this is the output I get. Did anyone encounter this issue before ? Or do you have any tips how to start using Gazebo with this robot files (mushr_description, mushr_base, mushr_realsense2) ?
![mushr_gazebo](https://user-images.githubusercontent.com/92849136/152763060-f307f17c-e95e-4401-9311-621de10cb4f5.png)

**Comment by schmittlema** [comment]
I have no experience with MuSHR and gazebo unfortunately. @planc509 would you have any ideas?

**Comment by palanc** [comment]
Hi @ArguiIm - Thanks for pointing this out. Gazebo requires all simulated links to have inertial params specified in the URDF, but some of the links associated with the front wheels didn't. I've added this to the URDF so the front wheels should now show up. Could you try the following:

1. Do a `git pull` in your _mushr_ repo
2. Re-compile your workspace using `catkin_make`
3. Try launching Gazebo again with the mushr model

Let us know how it goes.

**Reply by ArguiIm** [reply]
Hey, thank you so much !
This works perfectly ! I spent a good amount of time looking for the issue !

**Comment by ArguiIm** [comment]
Hey @planc509,
I would like to ask you about the controllers used in order to make mushr robot move.
Since I'm using Mushr in Gazebo, I try to implement some easy algorithms (like teleoperation with keyboard), but weirdly, the robot doesn't move correctly, instead, it jumps, and I was wondering if you have any clue about this issue ? 
I'll show how the robot reacts in the simulation. 

https://user-images.githubusercontent.com/92849136/158785331-f082abd9-e5e9-4c49-928a-44a64f61b077.mp4

**Comment by palanc** [comment]
Hi @ArguiIm - I'm not sure what's wrong just from the video. If you'd like to post your code for teleoperating the robot I can take a look.

**Reply by ArguiIm** [reply]
Hey, thank's a loot for your time.
All the code is in the git repo : https://github.com/ArguiIm/mushr_workspace
The code for teleoperating is in : src/racecar_gazebo/racecar_control/scripts/keyboard_teleop1.py

**Reply by palanc** [reply]
I think the issue stems from the fact that, because MuSHR doesn't support Gazebo, we haven't tuned the physical properties of the MuSHR URDF. Specifically, all of the inertial values in the URDF are just placeholder values. There could also be issues with the collision meshes. I don't know if this will get fixed any time soon because for now we only support physics simulation in MuJoCo (which is now free: [https://mujoco.org/](https://mujoco.org/)). If you'd like to try that out, check out our tutorial for setting up MuSHR in MuJoCo here:  [https://mushr.io/tutorials/mujoco/](https://mushr.io/tutorials/mujoco/). Sorry I don't have a direct fix for your problem.

**Reply by ArguiIm** [reply]
Hey, thank's for your time and reply?
I'll give it another try before going to MuJoCo, but thank's again.
